lib/apache_svn_url_parser.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+class ApacheSvnUrlParser < URLParser
+  private
+
+  def full_domain
+    'https://svn.apache.org/viewvc'
+  end
+
+  def tlds
+    %w(org)
+  end
+
+  def domain
+    'svn.apache'
+  end
+
+  def domain_regex
+    # match only the viewvc endpoint at the domain
+    "#{domain.split("/").first}\.(#{tlds.join('|')})\/viewvc"
+  end
+
+  def remove_domain
+    # find the matches for any github domain characters in the url string
+    # and replace only the first match incase we find a repo with something like github.com as the name
+    url.sub!(/(apache\.org\/(viewvc|repos))+?(:|\/)?/i, '')
+  end
+
+  def extractable_early?
+    false
+  end
+
+  def remove_extra_segments
+    # split the url by / and remove any empty sections
+    self.url = url.split('/').reject{ |s| s.strip.empty? }
+
+    # check to see if any repository subdirectories are included in the segments
+    # this parser is parsing SVN projects, so any common folders used for branching should trip this
+    # truncate the array of segments to stop once we hit a top level sub directory typically seen in SVN repos
+    # and return everything up to that point
+    #
+    # for example apache.org/viewvnc/myproject/subproject/tags/my-1.0.0-release should stop at myproject/subproject
+    # since the tags are just part of that repository
+    subdir_names = %w[trunk tags branches]
```

Comment:
WDYT about making this a constant?

Reply:
Updated in 4565f0f512c5967d20ab20cf98c89501ed648b54